cmd/sops/subcommand/exec/exec.go

```diff
@@ -134,6 +135,14 @@ func ExecWithEnv(opts ExecOpts) error {
 
 	env = append(env, opts.Env...)
 
+	if opts.SameProcess {
+		if opts.Background {
+			log.Fatal("background is not supported for same-process")
```

Comment:
I think this should also be checked in cmd/sops/main.go, so that the program exists with an error message and never gets to this point. (Please keep this check here so that if someone calls this function with an invalid combination, there's still some warning.)